src/bun.js/child_process.exports.js

```diff
@@ -1,3 +1,995 @@
+  );
+}
+
+function ERR_INVALID_ARG_VALUE(name, value, reason) {
+  return new Error(
+    `The value '${value}' is invalid for argument '${name}'. Reason: ${reason}`
+  );
+}
+
+// TODO: Add actual proper error implementation here
+function errnoException(err, name) {
+  return new Error(`Error: ${name}. Internal error: ${err.message}`);
+}
+
+//------------------------------------------------------------------------------
+// 7. Node stream polyfills
+//------------------------------------------------------------------------------
+Readable.prototype.on = function (event, listener) {
+  EventEmitter.prototype.on.call(this, event, listener);
+  if (event === "data") {
+    this._readableState.flowing = true;
+    this._read();
+  }
+};
+/**
+ * @param {ReadableStream} readableStream
+ * @param {{
+ *   highWaterMark? : number,
+ *   encoding? : string,
+ *   objectMode? : boolean,
+ *   signal? : AbortSignal,
+ * }} [options]
+ * @returns {Readable}
+ */
+export function newStreamReadableFromReadableStream(
+  readableStream,
+  options = {}
+) {
+  if (!isReadableStream(readableStream)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      "readableStream",
+      "ReadableStream",
+      readableStream
+    );
+  }
+
+  validateObject(options, "options");
+  const { highWaterMark, encoding, objectMode = false, signal } = options;
+
+  if (encoding !== undefined && !Buffer.isEncoding(encoding))
+    throw new ERR_INVALID_ARG_VALUE(encoding, "options.encoding");
+  validateBoolean(objectMode, "options.objectMode");
+
+  const reader = readableStream.getReader();
+
+  let closed = false;
+
+  const readable = new Readable({
+    objectMode,
+    highWaterMark,
+    encoding,
+    signal,
+
+    read() {
+      reader
+        .read()
+        .then((chunk) => {
+          if (chunk.done) {
+            // Value should always be undefined here.
+            readable.push(null);
+          } else {
+            readable.push(chunk.value);
+          }
+        })
+        .catch((error) => destroy(readable, error));
+    },
+
+    destroy(error, callback) {
+      function done() {
+        try {
+          callback(error);
+        } catch (error) {
+          // In a next tick because this is happening within
+          // a promise context, and if there are any errors
+          // thrown we don't want those to cause an unhandled
+          // rejection. Let's just escape the promise and
+          // handle it separately.
+          process.nextTick(() => {
+            throw error;
+          });
+        }
+      }
+
+      if (!closed) {
+        reader.cancel(error).then(done).catch(done);
+        return;
+      }
+      done();
+    },
+  });
+
+  reader.closed
+    .then(() => {
+      closed = true;
+    })
+    .catch((error) => {
+      closed = true;
+      destroy(readable, error);
+    });
+
+  return readable;
+}
+
+function isReadableStream(value) {
+  return value instanceof ReadableStream;
```

Comment:
Since the user may override `ReadableStream` at runtime, we should also assign it to a local variable at the top of the file similarly to other primordials

Do we need to also check that `value` is a non-null `object` (to avoid throwing due to `undefined` or otherwise)

Reply:
Good catch, probably is necessary to check those values explicitly